File: Level 2/영어 끝말잇기/solution.js
function solution(n, words) {
  const usedWords = new Set();
  let startChar = words[0][0];

  for (let i = 0; i < words.length; i++) {
    const word = words[i];
    if (usedWords.has(word) || word[0] !== startChar) {
      return [(i % n) + 1, Math.floor(i / n) + 1];
    }
    usedWords.add(word);
    startChar = word.at(-1);
  }

  return [0, 0];
}

File: Level 2/영어 끝말잇기/solution.py
def solution(n, words):
    used_words = set()
    start_char = words[0][0]

    for index, word in enumerate(words):
        if word in used_words or word[0] != start_char:
            return [index % n + 1, index // n + 1]
        used_words.add(word)
        start_char = word[-1]

    return [0, 0]
